### PR TITLE
Changed Docker image tagging to use semver tags for releases.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

This allows users to use image names like `netdata/netdata:1` or `netdata/netdata@1.29` and track the most up-to-date release that matches that version prefix.

Such usage is a common practice for projects using semantic versioning like we are.

##### Component Name

area/ci
area/packaging

##### Test Plan

The tag generations cript can be tested locally, it takes a single argument which is a version string as we normally use them (so `vX.Y.Z`) and outputs a list of version tags to push.

##### Additional Information

Short term, the `1` tag will be functionally identical to the `stable` tag. Long-term, we may eventually end up with a new major version of Netdata, at which point having the `1` tag to let users stick with the old version is important during the transition period.

### CODE REVIEW

**Review by knatsakis — APPROVED** [review]

**Review by vkalintiris — APPROVED** [review]

**Comment by Ferroin** [comment]
I am intentionally holding off merging this until we know that we will not have another v1.29.x patch release. It should only go in for v1.30.0 because it introduces a breaking change in our handling of Docker image version tags.

**Comment by Ferroin** [comment]
I just realized we can do this in a backwards-compatible way, and that doing so would actually make us more consistent with how other projects are actually doing things (they usually do in fact have a leading `v` in their version tags).

Given this, if I could get this re-approved, it can be merged now instead of having to wait for an indeterminate amount of time.

**Review by knatsakis — DISMISSED** [review]

File: packaging/docker/README.md
Comment (joelhans) [inline packaging/docker/README.md:105]:
```suggestion
  or a nightly build. This is what Docker will use by default if you do not specify a tag.
```

**Review by joelhans — COMMENTED** [review]
This looks good. You think most users who care about these tag options will know how to use them, or should we add a sentence or two, or a link, explaining to add the tag to the end of `netdata/netdata`?

**Comment by Ferroin** [comment]
> This looks good. You think most users who care about these tag options will know how to use them, or should we add a sentence or two, or a link, explaining to add the tag to the end of `netdata/netdata`?

In general, if a user does not know how to select specific tags for a Docker image, they should not be using Docker, so I’d say it’s beyond the scope of _our_ documentation to try and help them there.

I also think it may come off as a bit patronizing to people who do know if we go through explaining how to use Docker tags.

**Review by OdysLam — APPROVED** [review]

**Comment by joelhans** [comment]
> In general, if a user does not know how to select specific tags for a Docker image, they should not be using Docker,

:laughing: Fair enough!

**Review by joelhans — APPROVED** [review]
Docs LGTM.

This should still get a review/approval from >1 engineer.

**Review by kaskavel — APPROVED** [review]

**Pull request merged**
